### ISSUE TITLE
All coffee files are compiled, linted, source mapped and documentation regenerated each time a single coffee file is saved

### ISSUE THREAD

**Issue by davej** [ahmednuaman/radian#7]
The watch steps should only apply to the edited file(s) not to every single file.

```
>> File "assets/js/routes.coffee" changed.

Running "coffeelint:all" (coffeelint) task
>> 23 files lint free.

Running "coffee:dev" (coffee) task
File assets/js/app.js created.
File assets/js/app.js.map created.
File assets/js/config.js created.
File assets/js/config.js.map created.
File assets/js/controller/app-controller.js created.
File assets/js/app-controller.js.map created.
File assets/js/routes.js created.
File assets/js/routes.js.map created.
File assets/js/startup.js created.
File assets/js/startup.js.map created.

Running "docco:all" (docco) task
docco: Gruntfile.coffee -> docs/Gruntfile.html
docco: assets/js/app.coffee -> docs/app.html
docco: assets/js/config.coffee -> docs/config.html
docco: assets/js/controller/app-controller.coffee -> docs/app-controller.html
docco: assets/js/routes.coffee -> docs/routes.html
docco: assets/js/startup.coffee -> docs/startup.html
docco: crawler.coffee -> docs/crawler.html
docco: grunt/coffeelint.coffee -> docs/coffeelint.html
docco: grunt/contrib-coffee.coffee -> docs/contrib-coffee.html
docco: grunt/contrib-compass.coffee -> docs/contrib-compass.html
docco: grunt/contrib-copy.coffee -> docs/contrib-copy.html
docco: grunt/contrib-imagemin.coffee -> docs/contrib-imagemin.html
docco: grunt/contrib-requirejs.coffee -> docs/contrib-requirejs.html
docco: grunt/contrib-watch.coffee -> docs/contrib-watch.html
docco: grunt/docco.coffee -> docs/docco.html
docco: grunt/exec.coffee -> docs/exec.html
docco: grunt/express-server.coffee -> docs/express-server.html
docco: grunt/jade.coffee -> docs/jade.html
docco: grunt/karma.coffee -> docs/karma.html
docco: grunt/text-replace.coffee -> docs/text-replace.html
docco: server.coffee -> docs/server.html
docco: test/unit/karma.conf.coffee -> docs/karma.conf.html
docco: test/unit/test-main.coffee -> docs/test-main.html
Completed in 4.960s at Sun Dec 29 2013 22:34:00 GMT+0000 (GMT)
```

You can read about how to do this here: https://github.com/gruntjs/grunt-contrib-watch#compiling-files-as-needed

Also I think it's worth considering moving `docco` out of `watch` because it's not really necessary for development.

**Comment by ahmednuaman** [comment]
This is a nice technique, can we also do it for the SASS files too?

**Comment by ahmednuaman** [comment]
I've pulled the PR but it only seems to watch the first file that changes. I'm looking at the function code now trying to work out what's going on.

**Comment by davej** [comment]
Yes we could do it for CSS but since the sass files are compiled down to `styles.css` would it make any difference?

I was having the same issue (only watched the first file) but when I removed `docco` it worked fine. Are you sure `docco` is removed from the task list? It seems that Docco doesn't support `spawn: false` properly.

It is possibly related to this issue: https://github.com/DavidSouther/grunt-docco/issues/32

**Comment by ahmednuaman** [comment]
Possibly, I'll have a look. I need to rewrite the Yeoman generator first, so once I've done that I'll pop back on to this.

**Comment by ahmednuaman** [comment]
Merged 0db8e72 into master
